pyqt-apps/siriushla/widgets/label.py

```diff
@@ -106,14 +107,27 @@ def value_changed(self, new_value):
             self.setText(time)
             return
 
-        # If it is a version string, replace multiple whitespaces with a single one
+        # If it is a version string, replace multiple whitespaces with a
+        # single one
         if self._display_format_type == self.DisplayFormat.BSMPUDCVersion:
             version = new_value[:16] + " " + new_value[16:] \
                 if new_value is not None else ''
             version = " ".join(version.split())
             self.setText(version)
             return
 
+        # If it is a TI Frm. version string, convert to unsigned int and
+        # represent in hex
+        if self._display_format_type == self.DisplayFormat.TIFRMVersion:
+            new_value = (new_value & ((1 << 32) - 1)) \
```

Comment:
What do you think about implementing this as: `ctypes.c_uint32(new_value).value`? It seems clearer about the intent, and if anyone has to copy the logic elsewhere, it's harder to get wrong.

Reply:
done!